File: solutions/codechef/WEIGHTBL/WEIGHTBL_5.java
// DATE: 03/09/2022, 06:09:47
// PROBLEM NAME: Weight Balance
// PROBLEM URL: https://www.codechef.com/problems/WEIGHTBL
// PROBLEM DIFFICULTY RATTING: 1045
// STATUS: accepted
// TIME: 0.82
// MEMORY: 125.3M

/* package codechef; // don't place package name! */

import java.util.*;
import java.lang.*;
import java.io.*;

/* Name of the class has to be "Main" only if the class is public. */
class Codechef
{
	public static void main (String[] args) throws java.lang.Exception
	{
	    Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        sc.nextLine();
        for (int i = 0; i < t; i++){
            int[] array = Arrays.stream(sc.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();
            if (array[1]-array[0] >= array[2]*array[4] && array[1]-array[0] <= array[3]*array[4]){
                System.out.println(1);
            }else{
                System.out.println(0);
            }
        }
	}
}

File: solutions/codechef/WEIGHTBL/WEIGHTBL_5.c
// DATE: 26/09/2022, 08:32:18
// PROBLEM NAME: Weight Balance
// PROBLEM URL: https://www.codechef.com/problems/WEIGHTBL
// PROBLEM DIFFICULTY RATTING: 1045
// STATUS: accepted
// TIME: 0.03
// MEMORY: 5.2M

#include <stdio.h>

int main(void) {
    int t;
    scanf("%d", &t);
    int w1, w2, x1, x2, m;
    for (int i=0; i<t; i++){
        scanf("%d%d%d%d%d", &w1,&w2,&x1,&x2,&m);
        if (w2-w1 >= x1*m && w2-w1 <= x2*m){
            printf("1\n");
        }else{
            printf("0\n");
        }
    }
	return 0;
}



